R-package/R/lgb.cv.R

```diff
@@ -211,18 +211,42 @@ lgb.cv <- function(params = list(),
     callbacks <- add.cb(callbacks, cb.record.evaluation())
   }
 
-  # Check for early stopping passed as parameter when adding early stopping callback
+  # If early stopping was passed as a parameter in params(), prefer that to keyword argument
+  # early_stopping_rounds by overwriting the value in 'early_stopping_rounds'
   early_stop <- c("early_stopping_round", "early_stopping_rounds", "early_stopping", "n_iter_no_change")
-  if (any(names(params) %in% early_stop)) {
-    if (params[[which(names(params) %in% early_stop)[1]]] > 0) {
-      callbacks <- add.cb(callbacks, cb.early.stop(params[[which(names(params) %in% early_stop)[1]]], verbose = verbose))
-    }
-  } else {
-    if (!is.null(early_stopping_rounds)) {
-      if (early_stopping_rounds > 0) {
-        callbacks <- add.cb(callbacks, cb.early.stop(early_stopping_rounds, verbose = verbose))
+  early_stop_param_indx <- names(params) %in% early_stop
+  if (any(early_stop_param_indx)) {
+    first_early_stop_param <- which(early_stop_param_indx)[[1]]
+    first_early_stop_param_name <- names(params)[[first_early_stop_param]]
+    early_stopping_rounds <- params[[first_early_stop_param_name]]
+  }
+
+  # Did user pass parameters that indicate they want to use early stopping?
+  using_early_stopping_via_args <- !is.null(early_stopping_rounds)
+
+  # Cannot use early stopping with 'dart' boosting
+  if (identical(params$boosting, "dart")){
```

Comment:
Does it cover cases when `dart` is passed via [boosting aliases](https://lightgbm.readthedocs.io/en/latest/Parameters.html#boosting)?

Reply:
Seems we don't actually use boosting aliases in the R package at all currently. We will pass through anything you pass into `...` in `lgb.train()` or `lgb.cv()` but we have no code I see in the R package that explictly needs to access that value. I'll add them to this PR since it seems like it would be in scope.

Reply:
Ok I just pushed https://github.com/microsoft/LightGBM/pull/2443/commits/cab2bcdd011e85c4dda008e69acc2ac390b63674 which adds the boosting aliases. I also added a unit test in that commit which will check that the warning is generated for all the aliases.

I think this approach of centralizing the alias map is preferable to having the aliases hard-coded as string literals in multiple places. For now I only added the boosting-related ones as that's within the scope of this PR. Let me know what you think about the approach! If you and @Laurae2 are ok with it, we could do that for all the aliases (in a separate PR later)

Reply:
@jameslamb 
> we could do that for all the aliases (in a separate PR later)

I don't think we need a map for aliases of **all parameters**, since they are handled at cpp side. Only aliases of params which are used at R side should be there, I guess.

Reply:
Agree, sorry, I also mean "all aliases used in the R side"